core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/QueryDao.scala

```diff
@@ -283,7 +289,10 @@ private[r2dbc] class QueryDao(settings: R2dbcSettings, connectionFactory: Connec
           serId = row.get[Integer]("event_ser_id", classOf[Integer]),
           serManifest = row.get("event_ser_manifest", classOf[String]),
           writerUuid = "", // not need in this query
-          tags = Set.empty, // tags not fetched in queries (yet)
+          tags = row.get("tags", classOf[Array[String]]) match {
+            case null      => Set.empty[String]
+            case tagsArray => tagsArray.toSet
+          },
           metadata = readMetadata(row)))
 
   def eventsByPersistenceId(
```

Comment:
There is one more below in eventsByPersistenceId, maybe best to update that at the same time so that we don't forget. We don't use that yet because it's using the classic EventEnvelope, but I think we have to add another currentEventsByPersistenceId that is using the new envelope.

Reply:
Yeah, I started and went back on it when I realised that's old envelope API. I'll add another r2dbc-only events by pid for now.